services/device-registry-mongodb/src/main/java/org/eclipse/hono/deviceregistry/mongodb/Application.java

```diff
@@ -68,15 +72,27 @@ public void setAuthenticationService(final AuthenticationService service) {
         this.authService = service;
     }
 
+    /**
+     * Sets the notification sender to use.
+     *
+     * @param notificationSender The notification sender.
+     * @throws NullPointerException if notificationSender is {@code null}.
+     */
+    @Autowired
```

Comment:
the notification sender is optional in the JDBC registry. Is there a particular reason why it is mandatory here?

Reply:
The JDBC device registry has the `registry-adapter` and `registry-management` Spring profiles for enabling the functionality of serving as device registry for the adapters and/or for managing the device data.
The notification sender is only used in the device management related classes, hence its association with the `registry-management` Spring profile.

The MongoDB registry doesn't have such a way of disabling the device management functionality. Therefore the notification sender is mandatory.

Reply:
I do not quite see how the fact that we deploy a notification sender is related to the inner workings of the JDBC registry? Isn't the refactoring of the sender into its own verticle also about decoupling it from the components that emit the notifications? FMPOV the JDBC registry will still work perfectly fine if there is no other verticle deployed that listens to the event bus and publishes notifications to Kafka/AMQP, or am I mistaken?

Reply:
Right, I've made the notification sender mandatory now in the JDBC registry as well.

Reply:
My point was actually, that FMPOV the notification sender should be *optional* in the JDBC registry as well

Reply:
Looking at the MongoDB device registry, the messaging network configuration for sending auto-provisioning event messages is mandatory. That means the needed basic configuration for the notification sender is already there anyway (provided in the Kafka case the basic client configuration is done using `hono.kafka.commonClientConfig`, as recommended).
I don't see why we should make the notification sender optional in the MongoDB device registry then. Some useful (albeit non-essential) features in the other Hono components rely on the device registry notifications mechanism.

Regarding the JDBC device registry:
I've now reverted the last change - the notification sender is optional again - in the sense that it is tied to the `registry-management` Spring profile. Otherwise registry startup would fail if `registry-management` isn't set and no messaging network is configured.
In general, I wonder if it really makes sense to have the `registry-management` Spring profile, allowing the device registry to be started without it. Usage of that profile also means that device auto-provisioning is disabled. Together with the implications regarding the notification sender these effects on protocol adapter behaviour are quite non-intuitive.

Reply:
The original idea behind being able to selectively enable/disable parts of the JDBC registry functionality was that the tenants might be managed using another system, e.g. by means of custom resources in k8s. However, I guess we can assume that use case to be no longer relevant. So, FMPOV we can simply remove the arbitrary Spring profiles from the JDBC registry. This mechanism would not work anyway when migrating it to Quarkus. WDYT?

Reply:
I would also vote for removing the Spring profile. This should be done as part of a separate issue/PR then, I guess.